### PR TITLE
Server integration

### PR DESCRIPTION
**Pull request by ObsessiveOrange**
Merged Test back in, added integration test & resolved some server-client communication issues.

This also tests code from @loganga

Reviewers: @jshap70 @fahslaj

### CODE REVIEW

**Comment by julie-is-late** [comment]
there are still some files from `.gradle` in here. I'd just go ahead and add it to your gitignore. In the future, just be careful using `git add -A` as you never know when you might accidentally pick up something that has passwords in it.

**Comment by ObsessiveOrange** [comment]
This is exactly the same commits as the previous PR. It's just cherry picked to reduce noise from the sync to test. i've already added it to my .gitignore.

**Comment by ObsessiveOrange** [comment]
Figured out why this kept happening, even with it in the .gitignore. It was tracked previously at some point, and that caused it to keep tracking it, even when it supposed to ignore the entire .gradle folder. Should be fixed now.

File: src/main/java/websocket/models/requests/ProjectRevokePermissionsRequest.java [hunk 1]
```diff
@@ -16,20 +18,14 @@
...
-        return new Request("Project", "RevokePermissions", this,
-                (response) -> {
-                    System.out.println("Received project revoke permissions response: " + response);
-                },
-                () -> {
-                    System.out.println("Failed to send project revoke permissions request to the server.");
-                });
-    }
-
     public ProjectRevokePermissionsRequest(long projectID, long revokeUsername) {
```
Comment (loganga) [inline src/main/java/websocket/models/requests/ProjectRevokePermissionsRequest.java:21]: revokeUsername should be a String

**Review by julie-is-late — APPROVED** [review]
I went over this in person with Ben and we made a few changes. 
Looks good! :+1:

**Pull request merged**